### PR TITLE
Variable direction

### PR DESCRIPTION
**Pull request by Br3nda**
Distinguish between input and output variables. Showing on the variables#show page, a list of places that variable is calculated.

Added a new variable_scenarios page, to see all scenarios calculating that variable as the output

### CODE REVIEW

**Comment by dlouise64** [comment]
@Br3nda some travis and cc issue, I'll review again when done :)

**Comment by Br3nda** [comment]
image
image

File: app/models/scenario_variable.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@
 class ScenarioVariable < ApplicationRecord
   belongs_to :scenario
   belongs_to :variable
+  validates :direction, presence: true, inclusion: {
+    in: %w[input output], message: '%{value} is not a valid direction'
```
[the remark this answers is not in the document]
Comment (Br3nda) [reply app/models/scenario_variable.rb]:
```suggestion
    in: %w[input output], message: '%<value>s is not a valid direction'
```

**Comment by Br3nda** [comment]
@dlouise64 All green and ready to test again

File: spec/models/scenario_spec.rb [hunk 2]
```diff
@@ -24,24 +24,56 @@
...
+      scenario do
+        complicated_scenario.parse_variables!
+        expect(subject).to eq ['student_allowance__eligible_for_basic_grant']
+        complicated_scenario.parse_variables!
+        expect(subject).to eq ['student_allowance__eligible_for_basic_grant']
+      end
+    end
+    describe 'removes variables we no longer refer to' do
+      scenario do
+        FactoryBot.create :variable, name: 'hates_marshmallows'
```
Comment (dlouise64) [inline spec/models/scenario_spec.rb]: @Br3nda aww poor marshmallows 😆

**Pull request merged**